dolphin/api/v1/pools.py

```diff
@@ -12,19 +12,93 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+import six
+from webob import exc
+
+from dolphin import db, exception
+from dolphin.api import common_utils as common
 from dolphin.api.common import wsgi
+from dolphin.api.views import pools as pool_view
+from dolphin.i18n import _
 
 
 class PoolController(wsgi.Controller):
+    def _is_storage_valid(self, context, pool, storage_id):
+        """verify the storage and pool association  ."""
+        try:
+            if pool.storage_id != storage_id:
+                return False
+            else:
+                storage = db.storage_get(context, storage_id)
+                if not storage:
+                    return False
+                else:
+                    return True
+        except exception.StorageNotFound:
+            msg = _("storage  '%s' not found.") % storage_id
+            raise exc.HTTPNotFound(explanation=msg)
+
+    def _get_pools_search_options(self):
+        """Return pools search options allowed ."""
+        return ('name', 'status', 'id', 'storage_id')
```

Comment:
Will it be better to keep a list and then return the list? Just for code readability

Reply:
done